### PR TITLE
Migrate Vendor EFT bank information

### PR DESCRIPTION
**Pull request by jaymckinney-enavate**
This change enhances the GP cloud migration to include migrating Vendor EFT Bank information from GP to BC.

- Added SY06000 (Address EFT Master) and MC40200 (Currency Setup) staging tables.
- If a Vendor EFT Bank account is marked as Inactive, it will not be migrated.
- New SWIFT Codes will be created in BC if they don't already exist.
- New Currencies will be created in BC if they don't already exist.
- The Vendor's Preferred Bank Account Code will be assigned with the following logic: 
     - It is populated with the EFT Bank Account assigned to the Vendor's Remit To address in GP. 
     - If there is no active Remit To address, then it will assign it to the Vendor's Primary address. 
     - If there is no active Primary address, then leave the Preferred Bank Account Code empty.

### CODE REVIEW

File: Apps/W1/HybridGP/app/src/Migration/Vendors/GPVendorMigrator.codeunit.al
Comment (nikolakukrika) [inline Apps/W1/HybridGP/app/src/Migration/Vendors/GPVendorMigrator.codeunit.al]: It is better to split the IF statement to be like
IF not GPSY06000Table.INACTIVE then
   if VendorTable.Get(GPSY06000Table.CustomerVendor_ID) then
   
This way we can save one GET statement per GPSY06000Table record.  AL will always run all conditions in IF statement (it is bad, but it has always been like this).

Alternative is to set the key on the GPSY06000 on INACTIVE and to do GPSY06000Table.SetRange(Inactive, false) before findset.

To avoid the get on the vendor table you can add the flow field to the GPSY06000 table named Vendor Exists that is of type lookup on the vendor No. Ping me if you need help with defining the flow field. Then you can eliminate the both conditions with 
GPSY06000Table.SetRange(Inactive, false);
GPSY06000Table.SetRange("Vendor Exists", true);
if not GPSY06000Table.FindSet() then
exit;

File: Apps/W1/HybridGP/app/src/Migration/Vendors/GPVendorMigrator.codeunit.al
Comment (nikolakukrika) [inline Apps/W1/HybridGP/app/src/Migration/Vendors/GPVendorMigrator.codeunit.al]: IF GeneralLedgerSetup.Get() then

File: Apps/W1/HybridGP/app/src/Migration/Vendors/GPVendorMigrator.codeunit.al
Comment (nikolakukrika) [inline Apps/W1/HybridGP/app/src/Migration/Vendors/GPVendorMigrator.codeunit.al]: Minor - we do not suffix the records with table in the code. It is the existing coding style.

File: Apps/W1/HybridGP/app/src/Migration/Vendors/GPVendorMigrator.codeunit.al
Comment (nikolakukrika) [inline Apps/W1/HybridGP/app/src/Migration/Vendors/GPVendorMigrator.codeunit.al:464]: Minor - might be good to extract to locked labels so it can be reused if needed.

File: Apps/W1/HybridGP/app/src/codeunits/GPCloudMigration.codeunit.al
Comment (nikolakukrika) [inline Apps/W1/HybridGP/app/src/codeunits/GPCloudMigration.codeunit.al:141]: This will result in merge conflict between you and Todd. Pull and resolve before submitting the final iteration.

File: Apps/W1/HybridGP/app/src/tables/GPCompanyAdditionalSettings.table.al
Comment (nikolakukrika) [inline Apps/W1/HybridGP/app/src/tables/GPCompanyAdditionalSettings.table.al]: Clear is better, it will remove everything.

**Review by nikolakukrika — APPROVED** [review]
Great changes. I have left some comments.  Feel free to ping me if you want a second review or if you have any questions.

**Comment by jaymckinney-enavate** [comment]
Resubmitting pull request without conflicts

**Pull request closed without merging**
